core/src/progress_map.rs

```diff
@@ -188,12 +189,39 @@ impl ValidatorStakeInfo {
     }
 }
 
+pub const RETRANSMIT_BASE_DELAY_MS: u64 = 5_000;
+pub const RETRANSMIT_BACKOFF_CAP: u32 = 6;
+
+#[derive(Debug, Default)]
+pub struct RetransmitInfo {
+    pub retry_time: Option<Instant>,
+    pub retry_iteration: u32,
+}
+
+impl RetransmitInfo {
+    pub fn reached_retransmit_threshold(&self) -> bool {
+        let backoff = std::cmp::min(self.retry_iteration, RETRANSMIT_BACKOFF_CAP);
+        let backoff_duration_ms = 2_u64.pow(backoff) * RETRANSMIT_BASE_DELAY_MS;
+        self.retry_time
+            .map(|time| time.elapsed().as_millis() > backoff_duration_ms.into())
+            .unwrap_or(true)
+    }
+
+    pub fn increment_retry_iteration(&mut self) {
+        if self.retry_time.is_some() {
+            self.retry_iteration += 1;
+        }
+        self.retry_time = Some(Instant::now());
+    }
+}
+
 pub struct ForkProgress {
     pub is_dead: bool,
     pub fork_stats: ForkStats,
     pub propagated_stats: PropagatedStats,
     pub replay_stats: ReplaySlotStats,
     pub replay_progress: ConfirmationProgress,
+    pub retransmit_info: RetransmitInfo,
```

Comment:
nice, this way, this should get added and pruned in the same way as the rest of the slot based progress entries